api_data.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2017 nexB Inc. and others. All rights reserved.
+# http://nexb.com and https://github.com/nexB/vulnerablecode/
+# The VulnerableCode software is licensed under the Apache License version 2.0.
+# Data generated with VulnerableCode require an acknowledgment.
+# VulnerableCode is a trademark of nexB Inc.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with VulnerableCode or any VulnerableCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with VulnerableCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  VulnerableCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  VulnerableCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/vulnerablecode/ for support and download.
+
+import json
+from urllib.request import urlopen
+
+def data_cve_circl(name, version=None):
+    """
+    Outputs cve-ids, if any, related to a package.
+    Take as input, a package name, type, package version and
+    query cve-search' dataset for any reported vulnerabilities
+    """
+    url = f'https://cve.circl.lu/api/search/{name}'
+
+    if version:
+        url += f'/{version}'
+
+    raw_data = urlopen(url).read()
+    data = json.loads(raw_data)
+
+    return data
+
+def extract_fields(data, fields_names, version=False):
+    """
+    Extracts requested data fields using data generated by
+    cve-search' api. Takes as input data, fields requested
+    """
+    extracted_data = []
+
+    if version:
+        data = data
+    else:
+        data = data['data']
+
+    for item in data:
+        results = {}
```

Comment:
@tdruez You missed ``for item in data`` and oddly, it is storing just a single entry? i think we should let go off the dict. comprehension because we need a try get block incase of a ``KeyError`` which tends to pop up if a key is missing somewhere in the data. This shouldn't happen, since the data obviously follows a pattern. But, there is something fishy in the backend.